chirp/drivers/retevis_ha1g.py

```diff
@@ -1325,6 +1362,26 @@ def get_features(self):
             "DTCS->DTCS"]
         return rf
 
+    def validate_memory(self, mem):
+        msgs = []
+
+        if (chirp_common.in_range(mem.freq, [self._airband])
+                and not self.supports_airband()):
+            msgs.append(chirp_common.ValidationError(
+                    _('Frequency in this range not supported by firmware')))
+
+        if 'AM' in MODES:
+            if chirp_common.in_range(mem.freq,
+                                     [self._airband]) and mem.mode != 'AM':
+                msgs.append(chirp_common.ValidationError(
+                    _('Frequency in this range requires AM mode')))
+            if not chirp_common.in_range(mem.freq,
+                                         [self._airband]) and mem.mode == 'AM':
+                msgs.append(chirp_common.ValidationError(
+                    _('Frequency in this range must not be AM mode')))
```

Comment:
These should be warnings and let the coercion in `get_memory()` fix it. See the notes here:

https://chirpmyradio.com/projects/chirp/wiki/DevelopersDriverNotes#Implied-modes

Basically, the way you have it, users will have to create a single memory in the airband range with AM already set and apply that idempotently (i.e. with a paste or right-click-properties edit). If you change this to warning, then it will warn them that the mode has to change, but allow the edit to continue, and then `get_memory()` will do the right thing so they see the proper mode.